File: src/Pattern.Application/Services/Roles/Dtos/RoleDto.cs
using Pattern.Application.Services.Base.Dtos;

namespace Pattern.Application.Services.Roles.Dtos
{
    public class RoleDto : EntityDto<Guid>
    {
        public string Name { get; set; }
        public List<PermissionDto> Permissions { get; set; }
    }
}

File: reactjs/src/services/role/role.dto.ts
import { EntityDto } from "../base/base.dto";

export interface PermissionDto extends EntityDto<number> {
  name: string;
}

export interface RoleDto extends EntityDto<string> {
  name: string;
  permissions: PermissionDto[];
}

export interface CreateRoleDto {
  name: string;
  permissionIds: number[];
}

export interface UpdateRoleDto extends EntityDto<string> {
  name: string;
  permissionIds: number[];
}
